saleor/graphql/payment/mutations/transaction/transaction_initialize.py

```diff
@@ -121,6 +133,27 @@ def clean_app_from_payment_gateway(cls, payment_gateway: PaymentGatewayData) ->
             )
         return app
 
+    @classmethod
+    def clean_gift_card_payment_gateway_data(
+        cls, payment_gateway: PaymentGatewayData
+    ) -> None:
+        data = payment_gateway.data
+        if (
+            not data
+            or not isinstance(data, dict)
+            or "code" not in data
+            or not data["code"]
+            or not isinstance(data["code"], str)
```

Comment:
Suggestion: Maybe we should use pydantic for his validation. Quite a lot condiitons to meet